lib/src/utils/parsing/parsing_utils.dart

```diff
@@ -0,0 +1,321 @@
+import 'package:ledger_cardano/src/errors/invalid_data_reason.dart';
+import 'package:ledger_cardano/src/models/network.dart';
+import 'package:ledger_cardano/src/models/parsed_input.dart';
+import 'package:ledger_cardano/src/models/parsed_network.dart';
+import 'package:ledger_cardano/src/models/parsed_output.dart';
+import 'package:ledger_cardano/src/models/parsed_transaction.dart';
+import 'package:ledger_cardano/src/models/transaction.dart';
+import 'package:ledger_cardano/src/models/tx_input.dart';
+import 'package:ledger_cardano/src/models/tx_output.dart';
+import 'package:ledger_cardano/src/utils/constants.dart';
+import 'package:ledger_cardano/src/utils/utilities.dart';
+
+
+bool isArray(dynamic data) => data is List;
+
+
+
+ParsedTransaction parseTransaction(Transaction tx) {
+  final network = parseNetwork(tx.network);
+  
+  // inputs
+  validate(isArray(tx.inputs), InvalidDataReason.inputsNotArray.message);
+  final inputs = tx.inputs.map((inp) => parseTxInput(inp)).toList();
+
+  // outputs 
+  validate(isArray(tx.outputs), InvalidDataReason.outputsNotArray.message);
+  final outputs = tx.outputs.map((o) => parseTxOutput(o, tx.network)).toList();
+
+  // fee
+  final fee = parseCoin(tx.fee, InvalidDataReason.feeInvalid);
+
+  // ttl
+  final ttl = tx.ttl == null 
+    ? null
+    : parseUint64Str(tx.ttl, {}, InvalidDataReason.ttlInvalid);
```

Comment:
tx.ttl should be BigInt by default so there won't be a need to "parse". You can instead add some BigInt validation to make sure that the number is positive and bitLength is <= 64. (see `writeSerializedUint64` where you already do the same)